shared/pkg/converter/cyclonedx_to_syft.go

```diff
@@ -74,23 +77,43 @@ func saveSyftSBOMToFile(syftBOM syft_sbom.SBOM, outputSBOMFile string) error {
 }
 
 func GetCycloneDXSBOMFromFile(inputSBOMFile string) (*cdx.BOM, error) {
-	inputSBOM, err := getCycloneDXSBOMBytesFromFile(inputSBOMFile)
+	inputSBOMBytes, err := getCycloneDXSBOMBytesFromFile(inputSBOMFile)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get CycloneDX SBOM bytes from file: %v", err)
+		return nil, fmt.Errorf("failed to get SBOM bytes from file: %v", err)
 	}
-	inputFormat := DetermineCycloneDXFormat(inputSBOM)
-	input := formatter.New(inputFormat, inputSBOM)
-	// use the formatter
-	if err = input.Decode(inputFormat); err != nil {
-		return nil, fmt.Errorf("unable to decode input SBOM %s: %v", inputSBOMFile, err)
+
+	// Ensure input is converted to cyclonedx regardless of the
+	// input SBOM type.
+	r := bytes.NewReader(inputSBOMBytes)
+	sbom, format, err := syft.Decode(r)
+	if err != nil {
+		return nil, fmt.Errorf("failed to identify or decode file %s to recognised SBOM format: %w", inputSBOMFile, err)
 	}
 
-	cdxBOM, ok := input.GetSBOM().(*cdx.BOM)
-	if !ok {
-		return nil, fmt.Errorf("failed to cast input SBOM: %v", err)
+	// If we've been given cyclonedx as the input decode directly using
+	// cyclonedx-go instead of going through syft's intermediary struct
+	// otherwise we may lose some properties/metadata which syft doesn't
+	// understand. Other format's metadata is best effort so we'll use
+	// syfts intermediary struct and then use ToFormatModel to switch it to
+	// cdx.BOM.
+	var bom *cdx.BOM
+	cdxFormat := cdx.BOMFileFormatXML
+	switch format {
+	case syft.FormatByName("cyclonedxjson"):
+		cdxFormat = cdx.BOMFileFormatJSON
+		fallthrough
+	case syft.FormatByName("cyclonedx"):
```

Comment:
In the case of cyclonedxjson , you will decode it with the syft decoder. By the comment above you want to convert it using cdx.NewBOMDecoder. Maybe you did mean to put case syft.FormatByName("cyclonedx"), case syft.FormatByName("cyclonedxjson"): ?

Reply:
the "fallthough" from cyclonedxjson falls through to the cyclonedx case after changing the cdxFormat variable, it doesn't fallthrough to the syft FormatToModel case in the switch statement

Reply:
But it will never enter to the cyclonedx case if it entered to the cyclonedxjson case, no?

Reply:
"fallthrough" will results in the code flow moving into the next case regardless of the condition

Reply:
Oh I didn't know that.. thought it is just checking it  👍